src/auth/app/service.go

```diff
@@ -3,112 +3,72 @@ package app
 import (
 	"context"
 	"crypto/x509"
-
-	"github.com/consensys/quorum-key-manager/src/auth/api/middlewares"
-	"github.com/consensys/quorum-key-manager/src/auth/entities"
-	"github.com/consensys/quorum-key-manager/src/auth/manager"
+	"github.com/consensys/quorum-key-manager/pkg/app"
+	service "github.com/consensys/quorum-key-manager/src/auth"
+	"github.com/consensys/quorum-key-manager/src/auth/api/http_middlewares"
+	"github.com/consensys/quorum-key-manager/src/auth/api/manifest"
+	authtypes "github.com/consensys/quorum-key-manager/src/auth/entities"
 	"github.com/consensys/quorum-key-manager/src/auth/service/authenticator"
-	apikey "github.com/consensys/quorum-key-manager/src/infra/api-key/filesystem"
+	"github.com/consensys/quorum-key-manager/src/auth/service/roles"
+	entities2 "github.com/consensys/quorum-key-manager/src/entities"
 	"github.com/consensys/quorum-key-manager/src/infra/jwt"
-	"github.com/consensys/quorum-key-manager/src/infra/jwt/jose"
 	"github.com/consensys/quorum-key-manager/src/infra/log"
-	manifestreader "github.com/consensys/quorum-key-manager/src/infra/manifests/filesystem"
-	tls "github.com/consensys/quorum-key-manager/src/infra/tls/filesystem"
 	"github.com/justinas/alice"
-
-	"github.com/consensys/quorum-key-manager/pkg/app"
 )
 
-func RegisterService(a *app.App, logger log.Logger) error {
-	// Load configuration
-	cfg := new(Config)
-	err := a.ServiceConfig(cfg)
-	if err != nil {
-		return err
+func RegisterService(
+	ctx context.Context,
+	a *app.App,
+	logger log.Logger,
+	manifests map[string][]entities2.Manifest,
+	jwtValidator jwt.Validator,
+	apikeyClaims map[string]*authtypes.UserClaims,
+	rootCAs *x509.CertPool,
+) (*roles.Interactor, error) {
+	// Business layer
+	// TODO: Create authorizator service here
+
+	var authenticatorService *authenticator.Authenticator
+	if jwtValidator != nil || apikeyClaims != nil || rootCAs != nil {
+		authenticatorService = authenticator.New(jwtValidator, apikeyClaims, rootCAs, logger)
 	}
 
-	manifestReader, err := manifestreader.New(cfg.Manifest)
-	if err != nil {
-		return err
-	}
+	rolesService := roles.New(logger)
 
-	compositionMiddleware, err := createMiddlewares(cfg, logger)
-	if err != nil {
-		return err
+	// Service layer
+	manifestRolesHandler := manifest.NewRolesHandler(rolesService) // Manifest reading is synchronous, similar to a config file
```

Comment:
we are mixing service initialization with service state initialization in all services. I would rather do it in two independent steps. Most likely we should look for a common space for all manifestHandlers where the state of all services are managed, WDYT?

Reply:
moved